File: step3.py
import multiprocessing
import time

def task(name):
    print(f"Starting task {name}")
    time.sleep(2)
    print(f"Task {name} completed")

if __name__ == "__main__":
    start = time.time()

    p1 = multiprocessing.Process(target=task, args=("A",))
    p2 = multiprocessing.Process(target=task, args=("B",))

    p1.start()
    p2.start()

    p1.join()
    p2.join()

    end = time.time()
    print(f"Total time: {end - start:.2f} seconds")

File: step3.go
// Shows only threads
package main

import (
	"fmt"
	"os"
	"strconv"
	"time"
)

func cpuHeavyTask(id int) {
	start := time.Now()
	total := 0
	for i := 0; i < 10000000; i++ {
		total += i * i
	}
	duration := time.Since(start)
	fmt.Printf("Process %d finished in %.4f seconds\n", id, duration.Seconds())
}

func main() {
	if len(os.Args) < 2 {
		fmt.Println("Usage: ./worker <id>")
		return
	}
	id, _ := strconv.Atoi(os.Args[1])
	cpuHeavyTask(id)
}
